src/lib/wallet/GoodWalletClass.js

```diff
@@ -451,46 +435,49 @@ export class GoodWallet {
     })
   }
 
-  async syncTxWithBlockchain(startBlock) {
+  async syncTxWithBlockchain(fromBlock) {
     const lastBlock = await this.wallet.eth.getBlockNumber()
-    startBlock = Math.min(startBlock, lastBlock)
+    const startBlock = Math.min(fromBlock, lastBlock)
     const steps = range(startBlock, lastBlock, POKT_MAX_EVENTSBLOCKS)
+
     log.debug('Start sync tx from blockchain', {
       steps,
       startBlock,
       lastBlock,
     })
 
     try {
-      const chunks = chunk(steps, 10)
-
-      for (let chunk of chunks) {
-        const ps = chunk.map(async fromBlock => {
-          let toBlock = fromBlock + POKT_MAX_EVENTSBLOCKS
-
-          // await callback to finish processing events before updating lastEventblock
-          // we pass toBlock as null so the request naturally requests until the last block a node has,
-          // this is to prevent errors where some nodes for some reason still dont have the last block
-          if (toBlock >= lastBlock) {
-            toBlock = undefined
-          }
-
-          log.debug('sync tx step:', { fromBlock, toBlock })
-
-          const events = flatten(
-            await Promise.all([
-              this.pollSendEvents(toBlock, fromBlock),
-              this.pollReceiveEvents(toBlock, fromBlock),
-              this.pollOTPLEvents(toBlock, fromBlock),
-            ]),
-          )
-
-          this._notifyEvents(events, fromBlock)
-        })
+      await chunk(steps, 10).reduce(
```

Comment:
no. not at all. another example of useless refactoring that changes the logic!

Reply:
as I remember this may be done to get rid of await in for () loop. reduce creates series of promise.then()'s so they executed one each other. 100% correct solution would be creating an async iterator.

but yep it's overkill, I will restore as it is in master